src/anomalib/metrics/base.py

```diff
@@ -130,13 +155,43 @@ def update(self, batch: Batch, *args, **kwargs) -> None:
         Raises:
             ValueError: If batch is missing any required fields.
         """
+        self.__update_count += 1
         for key in self.fields:
             if getattr(batch, key, None) is None:
-                msg = f"Batch object is missing required field: {key}"
+                # We cannot update the metric if the batch is missing required fields,
+                # so we need to decrement the update count of the super class.
+                self._update_count -= 1  # type: ignore[attr-defined]
+                if not self.strict:
+                    # If not in strict mode, skip updating the metric but don't raise an error
+                    return
+                # otherwise, raise an error
+                if not hasattr(batch, key):
+                    msg = (
+                        f"Cannot update metric of type {type(self)}. Passed dataclass instance "
+                        f"is missing required field: {key}"
+                    )
+                else:
+                    msg = (
+                        f"Cannot update metric of type {type(self)}. Passed dataclass instance "
+                        f"does not have a value for field with name {key}."
+                    )
                 raise ValueError(msg)
+
         values = [getattr(batch, key) for key in self.fields]
         super().update(*values, *args, **kwargs)  # type: ignore[misc]
 
+    def compute(self) -> torch.Tensor:
+        """Compute the metric value.
+
+        If the metric has not been updated, and metric is not in strict mode, return None.
+
+        Returns:
+            torch.Tensor: Computed metric value or None.
+        """
+        if self._update_count == 0 and not self.strict:  # type: ignore[attr-defined]
+            return None
+        return super().compute()  # type: ignore[misc]
```

Comment:
Can you elaborate the types are ignored? `compute` always expected to return tensor?

Reply:
Without the ignore, mypy raises the following error:
```
src/anomalib/metrics/base.py:191: error: "compute" undefined in superclass  [misc]
```
Mypy does not recognize the parent class, because this class can be added to any Metric class, sort of like a mixin.